common/ferrostar/src/models.rs

```diff
@@ -59,6 +59,42 @@ impl From<Rect> for BoundingBox {
     }
 }
 
+/// The heading of the user/device.
+/// 
+/// Ferrostar prefers course over ground, but may use heading in some cases.
+#[derive(Clone, Copy, PartialEq, PartialOrd, Debug, uniffi::Record)]
+pub struct Heading {
+    /// The heading in degrees relative to true north.
+    pub true_heading: u16,
+    /// The maximum deviation in degrees between the reported heading and the true geomagnetic heading.
```

Comment:
Heading is something we'd need to manually calculate from Sensor data sdk(s) on android as far as I can tell. So I think for now this holds true. We could revise it if and when we integrate an android "device" heading.

Reply:
After looking at it a bit, Android does expose it via the raw sensor APIs. I would suggest not making any guarantees in the doctrines about its accuracy though. It will be highly hardware specific at any rate.